src/nn/tfjs/TfjsModel.js

```diff
@@ -490,16 +515,34 @@ export default class WebGLModel {
           filterW = operands[inputs[i++]].value[0];
           filterH = operands[inputs[i++]].value[0];
           activation = FuseFunctionMap.get(operands[inputs[i++]].value[0]);
-          if (op === OperationCode.AVERAGE_POOL_2D) {
-            output.assign(activation(
-                input.avgPool([filterH, filterW],
-                              [strideH, strideW],
-                              padding)));
+          if (this._model._backend==='WASM' && filterH===1 && filterW===1) {
+            let inputData;
+            const [N, H, W, C] = output.shape;
+            const [NI, HI, WI, CI] = input.shape;
+            if (padding === 'same') {
+              const paddingLeft = Math.floor(((W-1)*strideW + filterW - WI)/2);
+              const paddingRight = ((W-1)*strideW + filterW - WI) - paddingLeft;
+              const paddingTop = Math.floor(((H-1)*strideH + filterH - HI)/2);
+              const paddingBottom = ((H-1)*strideH + filterH - HI) - paddingTop;
+              inputData = input.pad([[0, 0], [paddingTop, paddingBottom],
+                                             [paddingLeft, paddingRight], [0, 0]]).dataSync();
+            } else {
+              inputData = input.dataSync()
```

Comment:
If we do not need to operate on inputData, can we avoid reading from tensor by using
`inputData = this._model._operands[inputs[0]];`

Reply:
+1. That's a good point.

Reply:
This is patch code for pooling operation with filter 1×1 since tfjs-wasm doesn't support this. And these code only run for `wasm` backend.

Reply:
Only for wasm should be fine.

Reply:
@NALLEIN , are you fine with this?